pkg/model/estimator/local/regressor/regressor.go

```diff
@@ -32,20 +32,16 @@ import (
 	"k8s.io/klog/v2"
 )
 
-const (
-	// TODO: determine node type dynamically
-	defaultNodeType int = 1
-)
-
 // ModelRequest defines a request to Kepler Model Server to get model weights
 type ModelRequest struct {
-	MetricNames  []string `json:"metrics"`
-	OutputType   string   `json:"output_type"`
-	EnergySource string   `json:"source"`
-	NodeType     int      `json:"node_type"`
-	Weight       bool     `json:"weight"`
-	TrainerName  string   `json:"trainer_name"`
-	SelectFilter string   `json:"filter"`
+	MetricNames  []string           `json:"metrics"`
+	OutputType   string             `json:"output_type"`
+	EnergySource string             `json:"source"`
+	NodeType     int                `json:"node_type"`
+	Weight       bool               `json:"weight"`
+	TrainerName  string             `json:"trainer_name"`
+	SelectFilter string             `json:"filter"`
+	Spec         config.MachineSpec `json:"spec"`
```

Comment:
minor: spec too generic imo 

```suggestion
	Spec         config.MachineSpec `json:"machine_spec"`
```

Reply:
After the PR in kepler-model-server https://github.com/sustainable-computing-io/kepler-model-server/pull/411 merged, I will update.